doc/source/ray-core/type-hint.rst

```diff
@@ -0,0 +1,121 @@
+Type Hints in Ray
+==================
+
+Ray provides comprehensive support for Python type hints with both remote functions and actors.
```

Comment:
I think we should say here that for actors, the type hinting is available since Ray 2.48, otherwise people who use older Ray versions will be confused. You can either do this in the main test or add a sphinx note.